frontend/controller/actions/identity.js

```diff
@@ -960,6 +969,43 @@ export default (sbp('sbp/selectors/register', {
     // And remove transient keys, which require a user password
     sbp('chelonia/clearTransientSecretKeys', [oldIEKid, oldIPKid, IEKid, IPKid])
   },
+  'gi.actions/identity/delete': async ({
+    contractID,
+    transientSecretKeys,
+    oldKeysAnchorCid
+  }: {
+    contractID: string,
+    transientSecretKeys: Secret<string[]>,
+    oldKeysAnchorCid: string
+  }) => {
+    const state = sbp('chelonia/contract/state', contractID)
+    if (!state?.attributes?.encryptedDeletionToken) {
+      throw new Error('Missing encrypted deletion token')
+    }
+
+    const transientSecretKeysEntries = transientSecretKeys.valueOf().map(
+      k => ([keyId(k), deserializeKey(k)])
+    )
+    const encryptedDeletionToken = state.attributes.encryptedDeletionToken
+
+    if (oldKeysAnchorCid) {
+      const IEK = transientSecretKeysEntries[0][1]
+      await processOldIekList(contractID, oldKeysAnchorCid, IEK)
+    }
+
+    const token = encryptedIncomingData(contractID, state, encryptedDeletionToken, NaN, Object.fromEntries(transientSecretKeysEntries), 'encryptedDeletionToken')
+    await sbp('chelonia/out/deleteContract', contractID, {
+      [contractID]: { token: new Secret(token.valueOf()) }
+    })
+  },
+  'gi.actions/identity/_ondeleted': async (contractID: string, state: Object) => {
+    const ourIdentityContractId = sbp('state/vuex/getters').ourIdentityContractId
+
+    if (contractID === ourIdentityContractId) {
+      await sbp('gi.actions/identity/logout')
+    }
```

Comment:
You said this code has been written in a way that it doesn't matter in what order the `_ondeleted` selectors are called.

Are you sure about this? Could you add a comment then if so explaining why it's safe to call `logout` and that doing so won't result in any conflicts with any other pending calls to `_ondeleted`?

Reply:
What else are we supposed to do if our identity contract has been deleted? In any case, the calls to `_ondeleted` are for handling state, and there is no more state when our identity contract is deleted.

Reply:
If our identity contract has been deleted, we probably also want to still clear out the old state for chatrooms and groups still, right? How can we do that if we're logged out?

Reply:
If our identity contract has been deleted, we can no longer use the app or log in, so clearing out the old state doesn't seem to be useful.

Reply:
I guess we can leave as is, but maybe a comment explaining why it's safe to do so would be helpful